File: 929 - Unique Email Addressed.js
/**
 * @param {string[]} emails
 * @return {number}
 */
var numUniqueEmails = function (emails) {
    let dot = /\./g, plus = /\+.*/;
    let s = new Set();
    for (let email of emails) {
        let at = email.split('@');
        let local = at[0], domain = at[1];
        local = local.replace(dot, '');
        local = local.replace(plus, '');
        s.add(local + '@' + domain);
    }
    return s.size;
};

let a =
    ["test.email+alex@leetcode.com", "test.e.mail+bob.cathy@leetcode.com", "testemail+david@lee.tcode.com"];
numUniqueEmails(a);

File: 929 - Unique Email Addressed.py
import re


class Solution:
    def numUniqueEmails(self, emails: "list[str]") -> int:
        e = set()
        for email in emails:
            at = email.find('@')
            local = email[:at]
            domain = email[at:]
            local = re.sub(r'\.', '', local)
            local = re.sub(r'\+.*', '', local)
            # print(local + domain)
            e.add(local + domain)
        return len(e)


s = Solution()
a = ["test.email+alex@leetcode.com",
     "test.e.mail+bob.cathy@leetcode.com", "testemail+david@lee.tcode.com"]
s.numUniqueEmails(a)
